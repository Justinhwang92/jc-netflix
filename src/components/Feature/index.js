import React from "react";
// Styles
import { Container, Title, SubTitle } from "./Feature.Styles";

export default function Feature({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Feature.Title = function FeatureTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Feature.SubTitle = function FeatureSubTitle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};
