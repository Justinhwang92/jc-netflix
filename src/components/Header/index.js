import React from "react";
import { Link as ReactRouterLink } from "react-router-dom";
// Styles
import { Background, ButtonLink, Container, Logo } from "./Header.Styles";

export const Header = ({ bg = true, children, ...restProps }) => {
  return bg ? <Background {...restProps}>{children}</Background> : children;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};
