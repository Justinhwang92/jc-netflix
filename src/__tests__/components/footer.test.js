import React from "react";
import { render } from "@testing-library/react";
import { Footer } from "../../components";

describe("<Footer />", () => {
  it("renders the <Footer /> with populated data", () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Questions? Contact us</Footer.Title>
        <Footer.Break />
        <Footer.Row>
          <Footer.Column>
            <Footer.Link href="#">FAQ</Footer.Link>
            <Footer.Link href="#">Investor Reations</Footer.Link>
            <Footer.Link href="#">Ways to Watch</Footer.Link>
            <Footer.Link href="#">Coporate Information</Footer.Link>
            <Footer.Link href="#">Netflix Originals</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Help Center</Footer.Link>
            <Footer.Link href="#">Jobs</Footer.Link>
            <Footer.Link href="#">Terms of Use</Footer.Link>
            <Footer.Link href="#">Contact Us</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Account</Footer.Link>
            <Footer.Link href="#">Reedem Gift Cards</Footer.Link>
            <Footer.Link href="#">Privacy</Footer.Link>
            <Footer.Link href="#">Speed Test</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Media Center</Footer.Link>
            <Footer.Link href="#">Buy Gift Cards</Footer.Link>
            <Footer.Link href="#">Cookie Preferences</Footer.Link>
            <Footer.Link href="#">Legal Notices</Footer.Link>
          </Footer.Column>
        </Footer.Row>
        <Footer.Break />
        <Footer.Text>Justin Coding - Netflix Clone Coding</Footer.Text>
      </Footer>
    );

    expect(getByText("Questions? Contact us")).toBeTruthy();
    expect(getByText("FAQ")).toBeTruthy();
    expect(getByText("Investor Reations")).toBeTruthy();
    expect(getByText("Ways to Watch")).toBeTruthy();
    expect(getByText("Coporate Information")).toBeTruthy();
    expect(getByText("Netflix Originals")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
