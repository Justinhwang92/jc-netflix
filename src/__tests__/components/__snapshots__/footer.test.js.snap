// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> with populated data 1`] = `
<div
  class="FooterStyle__Container-sc-a76zru-0 knAucQ"
>
  <p
    class="FooterStyle__Title-sc-a76zru-4 dTUVpJ"
  >
    Questions? Contact us
  </p>
  <div
    class="FooterStyle__Break-sc-a76zru-6 eZEIKa"
  />
  <div
    class="FooterStyle__Row-sc-a76zru-2 iIlfrP"
  >
    <div
      class="FooterStyle__Column-sc-a76zru-1 kcRQso"
    >
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        FAQ
      </a>
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Investor Reations
      </a>
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Ways to Watch
      </a>
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Coporate Information
      </a>
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="FooterStyle__Column-sc-a76zru-1 kcRQso"
    >
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Help Center
      </a>
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Jobs
      </a>
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Terms of Use
      </a>
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Contact Us
      </a>
    </div>
    <div
      class="FooterStyle__Column-sc-a76zru-1 kcRQso"
    >
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Account
      </a>
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Reedem Gift Cards
      </a>
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Privacy
      </a>
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Speed Test
      </a>
    </div>
    <div
      class="FooterStyle__Column-sc-a76zru-1 kcRQso"
    >
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Media Center
      </a>
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Buy Gift Cards
      </a>
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Cookie Preferences
      </a>
      <a
        class="FooterStyle__Link-sc-a76zru-3 drOdeI"
        href="#"
      >
        Legal Notices
      </a>
    </div>
  </div>
  <div
    class="FooterStyle__Break-sc-a76zru-6 eZEIKa"
  />
  <p
    class="FooterStyle__Text-sc-a76zru-5 iCCMJQ"
  >
    Justin Coding - Netflix Clone Coding
  </p>
</div>
`;
